### PR TITLE
Fix some typos

### PR DESCRIPTION
**Pull request by yuexiao-wang**
**- What I did**
Fix some typos when I read codes in docker

**- How I did it**

**- How to verify it**

**- Description for the changelog**

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]
LGTM

File: registry/auth.go [hunk 1]
```diff
@@ -285,7 +285,7 @@ func PingV2Registry(endpoint *url.URL, transport http.RoundTripper) (challenge.M
 		if pingVersion == v2Version {
 			// The version header indicates we're definitely
 			// talking to a v2 registry. So don't allow future
-			// fallbacks to the v1 protocol.
+			// fallback to the v1 protocol.
```
Comment (thaJeztah) [inline registry/auth.go]: I think `fallbacks` is correct here

File: image/image.go [hunk 2]
```diff
@@ -29,30 +29,30 @@ func IDFromDigest(digest digest.Digest) ID {
 
 // V1Image stores the V1 image configuration.
 type V1Image struct {
-	// ID a unique 64 character identifier of the image
-	ID string `json:"id,omitempty"`
+	// ID is a unique 64 character identifier of the image
+	ID              string `json:"id,omitempty"`
```
Comment (thaJeztah) [inline image/image.go]: Is this correct? Did you manually make these whitespace changes? running `gofmt` on this file gives me;

```patch
diff --git a/image/image.go b/image/image.go
index 75dc92e..58e5a7e 100644
--- a/image/image.go
+++ b/image/image.go
@@ -30,29 +30,29 @@ func IDFromDigest(digest digest.Digest) ID {
 // V1Image stores the V1 image configuration.
 type V1Image struct {
        // ID is a unique 64 character identifier of the image
-       ID              string `json:"id,omitempty"`
+       ID string `json:"id,omitempty"`
        // Parent id of the image
-       Parent          string `json:"parent,omitempty"`
+       Parent string `json:"parent,omitempty"`
        // Comment user added comment
-       Comment         string `json:"comment,omitempty"`
+       Comment string `json:"comment,omitempty"`
        // Created timestamp when image was created
-       Created         time.Time `json:"created"`
+       Created time.Time `json:"created"`
        // Container is the id of the container used to commit
-       Container       string `json:"container,omitempty"`
+       Container string `json:"container,omitempty"`
        // ContainerConfig is the configuration of the container that is committed into the image
        ContainerConfig container.Config `json:"container_config,omitempty"`
        // DockerVersion specifies version on which image is built
-       DockerVersion   string `json:"docker_version,omitempty"`
+       DockerVersion string `json:"docker_version,omitempty"`
        // Author of the image
-       Author          string `json:"author,omitempty"`
+       Author string `json:"author,omitempty"`
        // Config is the configuration of the container received from the client
-       Config          *container.Config `json:"config,omitempty"`
+       Config *container.Config `json:"config,omitempty"`
        // Architecture is the hardware that the image is build and runs on
-       Architecture    string `json:"architecture,omitempty"`
+       Architecture string `json:"architecture,omitempty"`
        // OS is the operating system used to build and run the image
-       OS              string `json:"os,omitempty"`
+       OS string `json:"os,omitempty"`
        // Size is the total size of the image including all layers it is composed of
-       Size            int64 `json:",omitempty"`
+       Size int64 `json:",omitempty"`
 }

 // Image stores the image configuration
@@ -65,7 +65,7 @@ type Image struct {
        OSFeatures []string  `json:"os.features,omitempty"`

        // rawJSON caches the immutable JSON associated with this image.
-       rawJSON    []byte
+       rawJSON []byte

        // computedID is the ID computed from the hash of the image config.
        // Not to be confused with the legacy V1 ID in V1Image.
@@ -113,17 +113,17 @@ func (img *Image) MarshalJSON() ([]byte, error) {
 // History stores build commands that were used to create an image
 type History struct {
        // Created timestamp for build point
-       Created       time.Time `json:"created"`
+       Created time.Time `json:"created"`
        // Author of the build point
-       Author        string `json:"author,omitempty"`
+       Author string `json:"author,omitempty"`
        // CreatedBy keeps the Dockerfile command used while building image.
-       CreatedBy     string `json:"created_by,omitempty"`
+       CreatedBy string `json:"created_by,omitempty"`
        // Comment is custom message set by the user when creating the image.
-       Comment       string `json:"comment,omitempty"`
+       Comment string `json:"comment,omitempty"`
        // EmptyLayer is set to true if this history item did not generate a
        // layer. Otherwise, the history item is associated with the next
        // layer in the RootFS section.
-       EmptyLayer    bool `json:"empty_layer,omitempty"`
+       EmptyLayer bool `json:"empty_layer,omitempty"`
 }

 // Exporter provides interface for exporting and importing images
```

**Review by thaJeztah — CHANGES_REQUESTED** [review]

**Comment by vdemeester** [comment]
@yuexiao-wang as @thaJeztah said, it's not properly `gofmt`'ed

```bash
11:13:52 These files are not properly gofmt'd:
11:13:52  - image/image.go
11:13:52 
11:13:52 Please reformat the above files using "gofmt -s -w" and commit the result.
```

File: registry/auth.go (hunk 1 above)
Comment (yuexiao-wang) [inline registry/auth.go]: Got it .

File: image/image.go (hunk 2 above)
Comment (yuexiao-wang) [inline image/image.go]: @thaJeztah  PTAL
Sorry，I ignored to check it by gofmt. 
It has done.

**Comment by yuexiao-wang** [comment]
@vdemeester PTAL
Done and thanks

File: image/image.go [hunk 3]
```diff
@@ -29,30 +29,24 @@ func IDFromDigest(digest digest.Digest) ID {
...
 	// Config is the configuration of the container received from the client
 	Config *container.Config `json:"config,omitempty"`
-	// Architecture is the hardware that the image is build and runs on
-	Architecture string `json:"architecture,omitempty"`
-	// OS is the operating system used to build and run the image
-	OS string `json:"os,omitempty"`
-	// Size is the total size of the image including all layers it is composed of
-	Size int64 `json:",omitempty"`
+
+	Architecture string `json:"architecture,omitempty"` // Architecture is the hardware that the image is build and runs on
```
Comment (thaJeztah) [inline image/image.go]: Why change all these? Looks to be inconsistent, because line 44 puts the documentation _above_, not _to the right_.

We may need to write down what our preferred notation is, but I'm not sure if should be in a "typo fix" pull request, because this is more than just a typo fix, and results in a lot more unrelated line-changes

Comment (yuexiao-wang) [inline image/image.go]: Line 44 is cut off when shown on GitHub. You can view it in split mode. 
Do I need to reopen a new PR for these change?

Comment (thaJeztah) [inline image/image.go]: No, I meant that with this change, we mix formatting to put the description _above_ some, and _to the right_ of others;

screen shot 2016-11-29 at 11 00 42

I would definitely prefer to have that in a separate PR if you think it's important (and why).

Even though I can see some formatting being slightly more readable, we should also take into account that docker gets a lot of pull requests; formatting should be such that it minimizes the chance of merge conflicts.

Comment (yuexiao-wang) [inline image/image.go]: I see and reverted this part

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM 🐸
